apps/sensors_test/src/main.c

```diff
@@ -413,10 +410,10 @@ config_sensor(void)
 
     dev = (struct os_dev *) os_dev_open("light0", OS_TIMEOUT_NEVER, NULL);
     assert(dev != NULL);
-    rc = tsl2561_init(dev, NULL);
-    if (rc) {
-        os_dev_close(dev);
-        goto err;
+
+    if (dev->od_flags & OS_DEV_F_STATUS_READY) {
+        rc= SYS_ENODEV;
+        console_printf("config_sensor %s init failed\n", dev->od_name);
```

Comment:
That is an intentional change so that even if initialization on one device fails the code goes ahead and tries to configure the same one and tries to initialize others.